internal/test/builder/builders.go

```diff
@@ -395,18 +395,19 @@ func (m *MachineDeploymentClassBuilder) Build() *clusterv1.MachineDeploymentClas
 }
 
 // InfrastructureMachineTemplateBuilder holds the variables and objects needed to build an InfrastructureMachineTemplate.
-// +kubebuilder:object:generate=false
 type InfrastructureMachineTemplateBuilder struct {
-	namespace  string
-	name       string
-	specFields map[string]interface{}
+	obj *unstructured.Unstructured
```

Comment:
The next step is to change the builder struct - here we change it to hold only the eventual object it will return when we call Build(). 

Here we need an InfrastructureMachineTemplate (which is type *`unstructured.Unstructured`

To express it slightly differently - obj should be the same as the return value in the Build() function for this type. Below we have:
```
func (i *InfrastructureMachineTemplateBuilder) Build() *unstructured.Unstructured {
```

Therefore obj should be *unstructured.Unstructured